### PR TITLE
Docs: Refactor and improve Standard/Extended API tables, font size and spacing fixes, improved syntax highlighting

### PR DESCRIPTION
**Pull request by brycelelbach**
* Refactor and improve the Standard API and Extended API overview tables
  * Overview tables have a short summary of the facility/header and use fewer
    columns.
  * The old `header_table.md` files have been folded into their section's
    overview file.
* Condense line spacing.
* Improve syntax highlighting (for some definition of improve).
* Increase size of code font.
* Fix some broken links.
* Update latest supported version of ICC and NVHPC.

### CODE REVIEW

File: docs/extended_api/synchronization_primitives.md [hunk 1]
```diff
@@ -1,48 +1,33 @@
...
 
-## Latches
+| [`cuda::barrier`]            | System-wide [`cuda::std::barrier`] multi-phase asynchronous thread coordination mechanism. `(class template)`  <br/><br/> 1.1.0 / CUDA 11.0 |
 
-| [`cuda::latch`]              | System-wide [`cuda::std::latch`] objects. `(class template)` |
+### Semaphores
 
-## Semaphores
+| [`cuda::counting_semaphore`] | System-wide [`cuda::std::counting_semaphore`] primitive for constraining concurrent access. `(class template)` <br/><br/> 1.1.0 / CUDA 11.0 |
+| [`cuda::binary_semaphore`]   | System-wide [`cuda::std::binary_semaphore`] primitive for mutual exclusion. `(class template)`                 <br/><br/> 1.1.0 / CUDA 11.0 |
```
Comment (griwes) [inline docs/extended_api/synchronization_primitives.md:18]: "System-wide cuda::std::X"? This is the extended API section, what is this trying to say?

File: docs/standard_api.md [hunk 2]
```diff
@@ -6,40 +6,33 @@ nav_order: 2
...
-| `<cuda/std/semaphore>` | 1.1.0 / CUDA 11.0 | [Standard API](https://en.cppreference.com/w/cpp/header/semaphore), [Extended API](./extended_api/synchronization_primitives/counting_semaphore.md) |
+Any Standard C++ header not listed below is omitted.
 
-## [Time Library](./standard_api/time_library.md)
+| [`<cuda/std/atomic>`]   | Atomic objects and operations (see also: [Extended API](./extended_api/synchronization_primitives/atomic.md)). <br/><br/> 1.0.0 / CUDA 10.2 |
+| [`<cuda/std/latch>`]    | Single-phase asynchronous thread-coordination mechanism (see also: [Extended API](./extended_api/synchronization_primitives/latch.md)). <br/><br/> 1.1.0 / CUDA 11.0 |
+| [`<cuda/std/barrier>`]  | Multi-phase asynchronous thread-coordination mechanism (see also: [Extended API](./extended_api/synchronization_primtives/barrier.md)). <br/><br/> 1.1.0 / CUDA 11.0 |
+| [`<cuda/std/semaphore>` | Primitives for constraining concurrent access (see also: [Extended API](./extended_api/synchronization_primitives/counting_semaphore.md)). <br/><br/> 1.1.0 / CUDA 11.0 |
```
Comment (griwes) [inline docs/standard_api.md:16]: Missing `]`.

File: docs/extended_api/shapes.md [hunk 3]
```diff
@@ -1,17 +1,10 @@
...
-| [`cuda::std::size_t`]    | Defines an extent of bytes. `(typedef)`                                            |
-| [`cuda::aligned_size_t`] | Defines an extent of bytes with a statically defined alignment. `(class template)` |
+| [`cuda::std::size_t`]    | Defines an extent of bytes. `(typedef)`                                            <br/><br/> 1.0.0 / CUDA 10.2 |
+| [`cuda::aligned_size_t`] | Defines an extent of bytes with a statically defined alignment. `(class template)` <br/><br/> 1.2.0 / CUDA 11.1 |
 
 
 [`cuda::std::size_t`]: https://en.cppreference.com/w/cpp/types/size_t
 
-[`cuda::aligned_size_t`]: ./shapes/aligned_size_t.md
+[`cuda::aligned_size_t`]: {{ "shapes/aligned_size_t.html" | relative_url }}
```
Comment (griwes) [inline docs/extended_api/shapes.md:9]: This ends up missing the `extended_api` path segment.

**Review by griwes — APPROVED** [review]
Fix the issues commented on below and this will be good to go.

File: docs/extended_api/synchronization_primitives.md (hunk 1 above)
Comment (brycelelbach) [reply docs/extended_api/synchronization_primitives.md:18]: It's an extended version of the thing that is scoped and thus can work across your entire system.

File: docs/standard_api.md (hunk 2 above)
Comment (brycelelbach) [reply docs/standard_api.md]: Fixed.

File: docs/extended_api/shapes.md (hunk 3 above)
Comment (brycelelbach) [reply docs/extended_api/shapes.md:9]: Fixed.

**Pull request merged**